extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/RoutingUtils.java

```diff
@@ -28,6 +28,9 @@ public static String getNormalizedAndDecodedPath(RoutingContext ctx) {
         if (normalizedPath.contains("?")) {
             return null;
         }
+        if (normalizedPath.indexOf('%') == -1) {
```

Comment:
I know that is unrelated to this issue but since we are here...
Uses String::indexOf('?')  since it is slightly better optimized and doesn't involve String matching , but single char

Reply:
WIll do the same for '?'